library/psa_crypto.c

```diff
@@ -2504,10 +2542,7 @@ psa_status_t psa_mac_sign_finish(psa_mac_operation_t *operation,
         operation->mac_size = 0;
     }
 
-    if (mac_size > operation->mac_size) {
-        memset(&mac[operation->mac_size], '!',
-               mac_size - operation->mac_size);
-    }
+    psa_wipe_tag_output_buffer(mac, status, mac_size, *mac_length);
```

Comment:
Why not keep the new code using `operation->mac_size` like the old code did?

Reply:
The code above zeros `operation->mac_size` on failure, which I presumed is still needed elsewhere, but I wanted to keep the logic the same, and let `psa_wipe_tag_input_buffer()` deal with the failure case, as it does in the other instances. (In the failure case it clears the whole buffer anyway)

Reply:
(just noting that in the failure case `psa_wipe_tag_input_buffer()` ignores the final parameter and wipes the whole of the buffer anyway)